cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,458 @@
+	if cut.SecurityContext != nil && cut.SecurityContext.AllowPrivilegeEscalation != nil {
+		containerSCC.PrivilegeEscalation = OK
+	}
+	containerSCC.PrivilegedContainer = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.Privileged != nil && *(cut.SecurityContext.Privileged) {
+		containerSCC.PrivilegedContainer = OK
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsUser != nil {
+		containerSCC.RunAsUserPresent = OK
+	}
+	containerSCC.ReadOnlyRootFilesystem = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.ReadOnlyRootFilesystem != nil && *cut.SecurityContext.ReadOnlyRootFilesystem {
+		containerSCC.ReadOnlyRootFilesystem = OK
+	}
+	containerSCC.RunAsNonRoot = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.RunAsNonRoot != nil && *cut.SecurityContext.RunAsNonRoot {
+		containerSCC.RunAsNonRoot = OK
+	}
+	if cut.SecurityContext != nil && cut.SecurityContext.SELinuxOptions != nil {
+		containerSCC.SeLinuxContextPresent = OK
+	}
+	return containerSCC
+}
+
+func updateCapabilities(cut *provider.Container, containerSCC *ContainerSCC) {
+	containerSCC.HaveDropCapabilities = NOK
+	if cut.SecurityContext != nil && cut.SecurityContext.Capabilities != nil {
+		var sliceDropCapabilities []string
+		for _, ncc := range cut.SecurityContext.Capabilities.Drop {
+			sliceDropCapabilities = append(sliceDropCapabilities, string(ncc))
+		}
+		sort.Strings(sliceDropCapabilities)
+
+		sort.Strings(requiredDropCapabilities)
+		if reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities) || reflect.DeepEqual(sliceDropCapabilities, dropAll) {
+			containerSCC.HaveDropCapabilities = OK
+		}
+
+		if checkContainCategory(cut.SecurityContext.Capabilities.Add, category2AddCapabilities) {
+			containerSCC.Capabilities = "category2"
+		} else {
+			if checkContainCategory(cut.SecurityContext.Capabilities.Add, category3AddCapabilities) {
+				containerSCC.Capabilities = "category3"
+			} else {
+				if len(cut.SecurityContext.Capabilities.Add) > 0 {
+					containerSCC.Capabilities = "category4"
+				} else {
+					logrus.Info("category is category1")
+					containerSCC.Capabilities = category1
+				}
+			}
+		}
+	} else {
+		containerSCC.Capabilities = category1
+	}
+}
+
+func AllVolumeAllowed(volumes []corev1.Volume) (r1, r2 OkNok) {
+	countVolume := 0
+	var value OkNok
+	value = NOK
+	for j := 0; j < len(volumes); j++ {
+		if volumes[j].HostPath != nil {
+			value = OK
+		}
+		if volumes[j].ConfigMap != nil {
+			countVolume++
+		}
+		if volumes[j].DownwardAPI != nil {
+			countVolume++
+		}
+		if volumes[j].EmptyDir != nil {
+			countVolume++
+		}
+		if volumes[j].PersistentVolumeClaim != nil {
+			countVolume++
+		}
+		if volumes[j].Projected != nil {
+			countVolume++
+		}
+		if volumes[j].Secret != nil {
+			countVolume++
+		}
+	}
+	if countVolume == len(volumes) {
+		return OK, value
+	}
+	return NOK, value
+}
+
+type CategoryID int
+
+const (
+	Undefined CategoryID = iota
+	CategoryID1
```

Comment:
Use these categories everywhere, remove all the string categories for instance "category1"